### DISCUSSION TITLE
Timer fired two times for multiple state having timer

### DISCUSSION THREAD

**Discussion by dennissetiawan** [indeedeng/iwf#321] [Q&A] [answered]
I find that if we have a workflow with state A -> B and each state have timer command.
When we move to state B the timer in state A is fired too.

<img width="1310" alt="image" src="https://github.com/indeedeng/iwf/assets/41095165/5bf9ff67-f90d-49bc-bee3-ac62526e25ea">

Is this behavior expected or should we do something to invalidate the state A timer?

**Comment by longquanzheng** [comment, marked as answer]
This is expected and the timer fired event  from state A will be a noop for the workflow. So you can just ignore it (Technically we could cancel the timer before moving to state B and this will save some workflow task for optimization  
https://github.com/indeedeng/iwf/issues/322)

**Reply by longquanzheng** [reply]
Because it’s noop you can see that there is workflow task completion without making any progress to the workflow.
